h/search/index.py

```diff
@@ -26,14 +26,18 @@ class Window(namedtuple('Window', ['start', 'end'])):
     pass
 
 
-def index(es, annotation, request, target_index=None):
+def index(es, annotation, request, target_index=None, refresh=False):
```

Comment:
In my previous experience with Elasticsearch, while the `refresh` parameter triggered a refresh immediately after the operation, it could still result in a race condition because the index operation would return before the refresh completed. As far as I can see from [our version’s docs](https://www.elastic.co/guide/en/elasticsearch/reference/1.6/docs-index_.html#index-refresh), this sounds like it could still be the case.

Do you know whether this has changed?

Reply:
Thanks for flagging this up. The docs [appear to make it clear](https://hyp.is/h3I3HtQ-Eee4dk8pq0KCow/www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) that the refresh should be synchronous in ES 6. Needs some investigation to understand whether this is (a) actually the case and (b) if so, does it apply to the version of ES we are using (1.6)?